system-addon/test/unit/lib/init-store.test.js

```diff
@@ -108,4 +108,50 @@ describe("initStore", () => {
       assert.calledWith(next, action);
     });
   });
+  describe("queueEarlyMessageMiddleware", () => {
+    it("should allow all local actions to go through", () => {
+      const action = {type: "FOO"};
+      const next = sinon.spy();
+
+      queueEarlyMessageMiddleware(store)(next)(action);
+
+      assert.calledWith(next, action);
+    });
+    it("should allow none EARLY_QUEUED_ACTIONS actions to main to go through", () => {
+      const action = ac.SendToMain({type: "FOO"});
+      const next = sinon.spy();
+
+      queueEarlyMessageMiddleware(store)(next)(action);
+
+      assert.calledWith(next, action);
+    });
+    it(`should line up EARLY_QUEUED_ACTIONS only let them go through after it receives the action from main`, () => {
+      EARLY_QUEUED_ACTIONS.forEach(actionType => {
+        const testStore = initStore({number: addNumberReducer});
+        const next = sinon.spy();
+        const action = ac.SendToMain({type: actionType});
+        const mergeStoreAction = ac.SendToContent({type: MERGE_STORE_ACTION}, 123);
+
+        // Early action should be added to the queue
+        queueEarlyMessageMiddleware(testStore)(next)(action);
+
+        assert.notCalled(next);
+        assert.equal(testStore._queue.length, 1);
```

Comment:
I think this test would be more interesting if multiple things got queued. Either the same action or different actions.